### PR TITLE
retrieve only relavent emails from mailtrap

### PR DESCRIPTION
**Pull request by bdsl**
I hope this should prevent repeats of some or all of the recent errors that we have seen from regression tests. By specifying the to_email of interest when we query emails in mailtrap we should significantly reduce the chance of not seeing the emails we need to check.

### CODE REVIEW

File: support/mailtrap.js [hunk 1]
```diff
@@ -22,10 +22,12 @@ async function mailtrapGet(path, responseType) {
  * Get the latest Mailtrap inbox message, if any.
  *
  * @param {int} count  Number of recent emails to get.
```
Comment (bdsl) [inline support/mailtrap.js:24]:
```suggestion
 * @param {int} count  Number of recent emails to get.
 * @see https://mailtrap.docs.apiary.io/#reference/message/apiv1inboxesinboxidmessagessearchpagelastid/get
```

File: support/mailtrap.js [hunk 2]
```diff
@@ -21,17 +21,26 @@ async function mailtrapGet(path, responseType) {
...
+async function getLatestMessages(toEmailAddress) {
+    if (typeof toEmailAddress !== 'string') {
+        throw new Error(`Expected toEmailAddress to be string, was ${typeof toEmailAddress}: ${toEmailAddress}`);
+    }
+
+    const params = new URLSearchParams({ search: toEmailAddress, page: '', last_id: '' });
+    const path = `/api/v1/inboxes/${process.env.MAILTRAP_INBOX_ID}/messages?${params.toString()}`;
     const messages = await mailtrapGet(path, 'json');
+
+    console.log(`got ${messages.length} message(s) from mailtrap at path ${path}`);
```
Comment (bdsl) [inline support/mailtrap.js:36]: Console log can be deleted before merge. Not committing now as I don't want to trigger another run.
```suggestion
    console.log(`got ${messages.length} message(s) from mailtrap at path ${path}`);
```

File: support/mailtrap.js [hunk 3]
```diff
@@ -21,17 +21,26 @@ async function mailtrapGet(path, responseType) {
...
  * @returns {array}     Up to {{count}} messages, if available.
  */
-async function getLatestMessages(count = 5) {
-    const path = `/api/v1/inboxes/${process.env.MAILTRAP_INBOX_ID}/messages?search=&page=&last_id=`;
+async function getLatestMessages(toEmailAddress) {
+    if (typeof toEmailAddress !== 'string') {
+        throw new Error(`Expected toEmailAddress to be string, was ${typeof toEmailAddress}: ${toEmailAddress}`);
+    }
+
+    const params = new URLSearchParams({ search: toEmailAddress, page: '', last_id: '' });
```
Comment (alihejazi) [inline support/mailtrap.js:32]: Might not be needed
```suggestion
    const params = new URLSearchParams({ search: toEmailAddress });
```

**Review by alihejazi — APPROVED** [review]

File: support/mailtrap.js (hunk 3 above)
Comment (bdsl) [reply support/mailtrap.js:32]: Right - I'll try and get passing before changing this.

**Review by NoelLH — APPROVED** [review]

**Comment by bdsl** [comment]
Merging with failed test run because it passed initially. Only failed because I decided to run it four times at once to see what would happen. Test runs interfere with each other I think because they all use some of the same accounts as defined in circle environment variables.

**Pull request merged**